File: Baekjoon/18258 - 큐 2/18258.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
    cin.tie(NULL);
	cin.sync_with_stdio(false);

    int n, k;
    string str;
    queue <int> q;

    cin >> n;
    for(int i = 0; i < n; i++) {
        cin >> str;
        if(str == "push") {
            cin >> k;
            q.push(k);
        }
        else if(str == "size") cout << q.size() << "\n";
        else if(str == "empty") cout << q.empty() << "\n";
        else if(str == "pop") {
            if(q.size() == 0) cout << -1 << "\n";
            else {
                cout << q.front() << "\n";
                q.pop();
            }
        }
        else if(str == "front") {
            if(q.size() == 0) cout << -1 << "\n";
            else cout << q.front() << "\n";
        }
        else if(str == "back") {
            if(q.size() == 0) cout << -1 << "\n";
            else cout << q.back() << "\n";
        }
    }
}

File: solution/18xxx/18258 - 큐 2/18258.py
import sys
input = sys.stdin.readline

queue = list()
left, right = 0, -1

for _ in range(int(input())):
    command = list(map(str, input().split()))

    if command[0] == 'push':
        queue.append(int(command[1]))
        right += 1
    elif command[0] == 'pop':
        print(queue[left] if right - left + 1 else -1)
        left += (1 if right - left + 1 else 0)
    elif command[0] == 'size':
        print(right - left + 1)
    elif command[0] == 'empty':
        print(0 if right - left + 1 else 1)
    elif command[0] == 'front':
        print(queue[left] if right - left + 1 else -1)
    elif command[0] == 'back':
        print(queue[right] if right - left + 1 else -1)